yarn-project/pxe/src/simulator_oracle/index.ts

```diff
@@ -316,57 +312,64 @@ export class SimulatorOracle implements DBOracle {
-    );
+    const indexes = await this.db.getTaggingSecretsIndexesAsRecipient(appTaggingSecrets);
+    return appTaggingSecrets.map((secret, i) => new IndexedTaggingSecret(secret, indexes[i]));
   }
 
   /**
-   * Synchronizes the logs tagged with the recipient's address and all the senders in the addressbook.
+   * Synchronizes the logs tagged with scopes addresses and all the senders in the addressbook.
    * Returns the unsynched logs and updates the indexes of the secrets used to tag them until there are no more logs to sync.
    * @param contractAddress - The address of the contract that the logs are tagged for
    * @param recipient - The address of the recipient
    * @returns A list of encrypted logs tagged with the recipient's address
    */
   public async syncTaggedLogs(
     contractAddress: AztecAddress,
-    recipient: AztecAddress,
-  ): Promise<TxScopedEncryptedL2NoteLog[]> {
-    // Ideally this algorithm would be implemented in noir, exposing its building blocks as oracles.
-    // However it is impossible at the moment due to the language not supporting nested slices.
-    // This nesting is necessary because for a given set of tags we don't
-    // know how many logs we will get back. Furthermore, these logs are of undetermined
-    // length, since we don't really know the note they correspond to until we decrypt them.
-
-    // 1. Get all the secrets for the recipient and sender pairs (#9365)
-    let appTaggingSecrets = await this.#getAppTaggingSecretsForSenders(contractAddress, recipient);
-
-    const logs: TxScopedEncryptedL2NoteLog[] = [];
-    while (appTaggingSecrets.length > 0) {
-      // 2. Compute tags using the secrets, recipient and index. Obtain logs for each tag (#9380)
-      const currentTags = appTaggingSecrets.map(taggingSecret => taggingSecret.computeTag());
-      const logsByTags = await this.aztecNode.getLogsByTags(currentTags);
-      const newTaggingSecrets: IndexedTaggingSecret[] = [];
-      logsByTags.forEach((logsByTag, index) => {
-        // 3.1. Append logs to the list and increment the index for the tags that have logs (#9380)
-        if (logsByTag.length > 0) {
-          logs.push(...logsByTag);
-          // 3.2. Increment the index for the tags that have logs (#9380)
-          newTaggingSecrets.push(
-            new IndexedTaggingSecret(appTaggingSecrets[index].secret, recipient, appTaggingSecrets[index].index + 1),
-          );
-        }
-      });
-      // 4. Consolidate in db and replace initial appTaggingSecrets with the new ones (updated indexes)
-      await this.db.incrementTaggingSecretsIndexes(
-        newTaggingSecrets.map(secret => new TaggingSecret(secret.secret, recipient)),
+    maxBlockNumber: number,
+    scopes?: AztecAddress[],
+  ): Promise<Map<string, TxScopedEncryptedL2NoteLog[]>> {
+    const recipients = scopes
+      ? scopes
+      : (await this.db.getCompleteAddresses()).map(completeAddress => completeAddress.address);
```

Comment:
Is this what empty scopes implies? That all accounts are available?

Reply:
Yes! That's how it currently works for everything else too